packetnetworking/utils.py

```diff
@@ -301,39 +296,47 @@ def get_lshw_info():
     return json.loads(get_output(["lshw", "-json", "-disable", "fb"]).decode())
 
 
+def get_devices_info():
+    path = "/sys/class/net/"
+    nics = []
+    ifaces = get_output(["ls", path]).decode().split()
+    for iface in ifaces:
+        try:
+            driver_path = (
+                get_output(["readlink", path + iface + "/device/driver"])
+                .decode()
+                .strip()
+            )
+            driver = get_output(["basename", driver_path]).decode().strip()
+        except subprocess.CalledProcessError:
+            continue
+
+        serial = get_output(["cat", path + iface + "/address"]).decode().strip()
```

Comment:
Hmm I think this should all be done as one "script" (the contents passed to get_output or similar) or just use python's built in functions for traversing directories, reading links and reading files :D.